gui/elements/qt/_dimensions.py

```diff
@@ -69,17 +75,17 @@ def update_slider(self, axis, max_axis_length):
             # slider.setTickInterval(tick_interval)
             slider.setSingleStep(1)
 
-            def value_changed():
-                self.viewer.dimensions.indices[axis] = slider.value()
-                self.viewer.dimensions._need_redraw = True
-                self.viewer.dimensions._update()
-
-            slider.valueChanged.connect(value_changed)
-
             grid.addWidget(slider, row, 0)
             self.sliders.append(slider)
         else:
             slider = slider.widget()
 
+        slider.valueChanged.connect(lambda value, axis=axis: self._slider_value_changed(value, axis))
```

Comment:
Why `axis=axis`?

Reply:
@kne42 because of this, a common Python gotcha that I too discovered quite late!

https://stackoverflow.com/a/10452819/

Reply:
But isn’t that only an issue if `axis` is changed? I don’t think that’s happening here.

Reply:
I presume @sofroniewn did not write this to show off and actually encountered a problem that this fixes? =) The connection is long-lived so the lambda could be called basically at any moment, and the local variable `axis` might be whatever at that point. I think?

Reply:
From my understanding of [this StackOverflow thread](https://stackoverflow.com/questions/2295290/what-do-lambda-function-closures-capture), it binds by both name **and** scope. This means that it is not necessary to force binding like this as long as the desired variable is not changed before the end of scope, which in this case would be the function.

This seems to cause problems mostly because people expect `while`- and `for`-loops to have their own scopes or something.

Reply:
Ah, I see what you're getting at. Took me a while! 😅 Well, you're probably right, I'll just let @sofroniewn weigh in at this point. 😊

Reply:
I was probably pattern matching to something else I had seen (some stackover flow post) without fully appreciating why it was needed (maybe they had required it for the reasons @jni mentions). In this case what @kne42 is saying makes sense and that function won't get called elsewhere, so I have removed the `axis=axis` and can verify that it works just fine!

Thanks for flagging and the links to the posts - definitely learnt something new about lambdas!

Not sure if this has implications for how the lambdas are defined in `_qt/_base_layer` or `_qt/_layer_buttons` - but maybe we wait to address those in a future PR